Regression: Panic when drawing canvas.Rectangle without FillColor in 2.4.0

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

I have a working application built in 2.3.5 and after upgrading to 2.3.6 ( develop ) I get a panic when I draw some elements on screen using a canvas.Rectangle without a FillColor

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0xdd95cb]

goroutine 21 [running, locked to thread]:
fyne.io/fyne/v2/internal/painter/gl.getFragmentColor({0x0, 0x0})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/painter/gl/draw.go:392 +0x1b
fyne.io/fyne/v2/internal/painter/gl.(*painter).drawRectangle(0xc71a000, 0xc4928c0, {0x40800000, 0x40800000}, {0x43fa0000, 0x43fa0000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/painter/gl/draw.go:148 +0x6e1
fyne.io/fyne/v2/internal/painter/gl.(*painter).drawObject(0xc71a000, {0x16f81ec, 0xc4928c0}, {0x40800000, 0x40800000}, {0x43fa0000, 0x43fa0000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/painter/gl/draw.go:81 +0x1c1
fyne.io/fyne/v2/internal/painter/gl.(*painter).Paint(0xc71a000, {0x16f81ec, 0xc4928c0}, {0x40800000, 0x40800000}, {0x43fa0000, 0x43fa0000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/painter/gl/painter.go:93 +0x78
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).paint.func1(0xc500690, {0x40800000, 0x40800000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/canvas.go:293 +0x160
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree.func1({0x16f81ec, 0xc4928c0}, {0x40800000, 0x40800000}, {0x0, 0x0}, {0x4f000000, 0x4f000000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/common/canvas.go:503 +0x1d0
fyne.io/fyne/v2/internal/driver.walkObjectTree({0x16f81ec, 0xc4928c0}, 0x0, {0x16f81b8, 0xc500600}, {0x40800000, 0x40800000}, {0x0, 0x0}, {0x4f000000, ...}, ...)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/util.go:169 +0x244
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/util.go:176
fyne.io/fyne/v2/internal/driver.walkObjectTree({0x16f81b8, 0xc500600}, 0x0, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x4f000000, ...}, ...)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/util.go:190 +0x37e
fyne.io/fyne/v2/internal/driver.WalkVisibleObjectTree({0x16f81b8, 0xc500600}, 0xc715e5c, 0xc715e48)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/util.go:134 +0x80
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree(0xc49a210, 0xc4890e0, 0xc715ec4, 0xc715eb0)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/common/canvas.go:524 +0x112
fyne.io/fyne/v2/internal/driver/common.(*Canvas).WalkTrees(0xc49a210, 0xc715ec4, 0xc715eb0)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/common/canvas.go:436 +0x3a
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).paint(0xc49a210, {0x43fa0000, 0x43fa0000})
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/canvas.go:309 +0xf1
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/loop.go:207 +0x97
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0xc51a900, 0xc715f14)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/window.go:917 +0x4c
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0xc492780, 0xc51a900)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/loop.go:198 +0x54
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame(0xc492780)
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/loop.go:106 +0x127
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1()
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/loop.go:254 +0x1ac
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread in goroutine 1
        C:/Users/roffe/go/pkg/mod/fyne.io/fyne/v2@v2.3.6-0.20230815105540-d225e4eb2a9e/internal/driver/glfw/loop.go:230 +0x102
exit status 2
```

### How to reproduce

Run example code

### Screenshots

_No response_

### Example code

```go
package main

import (
	"image/color"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/canvas"
	"fyne.io/fyne/v2/container"
)

func main() {
	a := app.New()
	w := a.NewWindow("Crash Test")
	w.Resize(fyne.NewSize(500, 500))

	rect := &canvas.Rectangle{StrokeColor: color.RGBA{0x80, 0x80, 0x80, 0x80}, StrokeWidth: 1}
	rect.Resize(fyne.NewSize(100, 100))

	w.SetContent(
		container.NewStack(rect),
	)
	w.ShowAndRun()
}
```

### Fyne version

2.3.6

### Go compiler version

1.21

### Operating system and version

Windows 11

### Additional Information

_No response_

--- a/internal/painter/gl/draw.go
+++ b/internal/painter/gl/draw.go
@@ -389,6 +389,9 @@ func roundToPixelCoords(size fyne.Size, pos fyne.Position, pixScale float32) (fy
 
 // Returns FragmentColor(red,green,blue,alpha) from fyne.Color
 func getFragmentColor(col color.Color) (float32, float32, float32, float32) {
+	if col == nil {
+		return 0, 0, 0, 0
+	}
 	r, g, b, a := col.RGBA()
 	if a == 0 {
 		return 0, 0, 0, 0

--- a/internal/painter/gl/draw_test.go
+++ b/internal/painter/gl/draw_test.go
@@ -4,6 +4,7 @@
 package gl
 
 import (
+	"image/color"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -11,6 +12,30 @@ import (
 	"fyne.io/fyne/v2"
 	"fyne.io/fyne/v2/canvas"
 )
+
+func TestGetFragmentColor(t *testing.T) {
+	var c color.Color
+
+	r, g, b, a := getFragmentColor(c)
+	assert.Equal(t, float32(0), r)
+	assert.Equal(t, float32(0), g)
+	assert.Equal(t, float32(0), b)
+	assert.Equal(t, float32(0), a)
+
+	c = color.NRGBA{R: 0x0, G: 0x66, B: 0x99, A: 0xff}
+	r, g, b, a = getFragmentColor(c)
+	assert.Equal(t, float32(0), r)
+	assert.Equal(t, float32(0.4), g)
+	assert.Equal(t, float32(0.6), b)
+	assert.Equal(t, float32(1), a)
+
+	c = color.NRGBA{R: 0x0, G: 0x66, B: 0x99, A: 0x99}
+	r, g, b, a = getFragmentColor(c)
+	assert.Equal(t, float32(0), r)
+	assert.Equal(t, float32(0.3999898), g)
+	assert.Equal(t, float32(0.59998477), b)
+	assert.Equal(t, float32(0.6), a)
+}
 
 func TestInnerRect_Stretch(t *testing.T) {
 	pos := fyne.NewPos(10, 10)

